src/architecture.c

```diff
@@ -419,19 +419,58 @@ static void jump(arch_core* core)
 
 static void branch(arch_core* core)
 {
+	arch_addr jmp_addr;
 	switch (core->regs.ir.opcode)
 	{
 		case BEQ:
+			if (((cond_imm_data)core->regs.ir.data).breg == ((cond_imm_data)core->regs.ir.data).dreg)
+			{
+				jmp_addr = ((cond_imm_data) core->regs.ir.data).addr;
+				core->regs.pc = jmp_addr;
+				return;
+			}
+			else {return;} 
```

Comment:
This code looks good. Just a note: you can just replace `break` with `return` instead of having an else statement return (it will return whether or not the if condition is true):

```c

case BEQ:
   if (((cond_imm_data)core->regs.ir.data).breg == ((cond_imm_data)core->regs.ir.data).dreg)
   {
      jmp_addr = ((cond_imm_data) core->regs.ir.data).addr;
      core->regs.pc = jmp_addr;
   }
   return;
```